sql/sql_acl.cc

```diff
@@ -11166,6 +11166,21 @@ acl_authenticate(THD *thd, uint com_change_user_pkt_len)
       general_log_print(thd, command, (char*) "%s@%s on %s",
                         mpvio.auth_info.user_name, mpvio.auth_info.host_or_ip,
                         mpvio.db.str ? mpvio.db.str : (char*) "");
+
+    if (log_legacy_user && opt_legacy_user_identifier &&
+        strstr(mpvio.auth_info.user_name, opt_legacy_user_identifier))
+    {
+      /* Log the legacy user */
+      char buffer[256];
+      snprintf(buffer, 256, "LEGACY_USER: %s@%s on %s",
+               mpvio.auth_info.user_name,
+               mpvio.auth_info.host_or_ip,
+               mpvio.db.str ? mpvio.db.str : (char*) "");
+      bool save_enable_slow_log = thd->enable_slow_log;
+      thd->enable_slow_log = true;
```

Comment:
Yeah that seems weird. We are actually piggyback the slow-log for the connection users. The `enable_slow_log` is set in the `dispatch_command` as it is associated with queries. So during connection auth, the variable isn't supposed to be set yet. The file path has a default value. Whether the file path is valid or not applies to all slow logging scenarios.